tracer/src/Datadog.Trace/Util/DbCommandCache.cs

```diff
@@ -97,13 +97,23 @@ public static TagsCacheItem GetTagsFromDbCommand(IDbCommand command)
 
         private static TagsCacheItem ExtractTagsFromConnectionString(string connectionString)
         {
-            // Parse the connection string
-            var builder = new DbConnectionStringBuilder { ConnectionString = connectionString };
-
-            return new TagsCacheItem(
-                dbName: GetConnectionStringValue(builder, "Database", "Initial Catalog", "InitialCatalog"),
-                dbUser: GetConnectionStringValue(builder, "User ID", "UserID"),
-                outHost: GetConnectionStringValue(builder, "Server", "Data Source", "DataSource", "Network Address", "NetworkAddress", "Address", "Addr", "Host"));
+            try
+            {
+                // Parse the connection string
+                var builder = new DbConnectionStringBuilder { ConnectionString = connectionString };
+
+                // Extract the tags
+                return new TagsCacheItem(
+                    dbName: GetConnectionStringValue(builder, "Database", "Initial Catalog", "InitialCatalog"),
+                    dbUser: GetConnectionStringValue(builder, "User ID", "UserID"),
+                    outHost: GetConnectionStringValue(builder, "Server", "Data Source", "DataSource", "Network Address", "NetworkAddress", "Address", "Addr", "Host"));
+            }
+            catch (Exception)
+            {
+                // DbConnectionStringBuilder can throw exceptions if the connection string is invalid (as seen in Error Tracking)
```

Comment:
nit: I would just say this
```suggestion
                // DbConnectionStringBuilder can throw exceptions if the connection string is invalid
```